### PR TITLE
Enable editing student's personal info

### PR DESCRIPTION
**Pull request by eduardotru**
Enable admin and super admin to edit student's personal information.
This PR also fixes a bug which didn't allow students to be registered due to a typo.

Tests: Modify a user and see the changes saved in the DB
Screenshot from 2020-04-02 15-49-43
Screenshot from 2020-04-02 15-49-48

### CODE REVIEW

File: app/controllers/students_controller.rb [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
     authorize @student
   end
 
-  def new
-    @user = User.new
+  def edit
+  end
+
+  def update
+    params.permit!
```
Comment (ReneSG) [inline app/controllers/students_controller.rb:19]: Please do not use `params.permit!` as it whitelist all the parameters and could lead to vulnerabilities. (https://stackoverflow.com/questions/52374655/ruby-on-rails-when-to-use-params-permit-and-how-to-replace-it)

Instead use strong parameters (https://edgeapi.rubyonrails.org/classes/ActionController/StrongParameters.html)

Comment (eduardotru) [reply app/controllers/students_controller.rb:19]: Done

File: app/controllers/students_controller.rb [hunk 2]
```diff
@@ -28,4 +33,28 @@ def clean(headers)
...
 
+  def set_student
+    @student = Student.find(params[:id])
+  end
+
+  def authorize_user
+    authorize User
+  end
+
+  def student_params
```
Comment (ReneSG) [inline app/controllers/students_controller.rb:44]: Make this private please.

Comment (eduardotru) [reply app/controllers/students_controller.rb:44]: It already is private, line 29

**Pull request merged**